spec/features/reviewing_step_by_steps_spec.rb

```diff
@@ -55,6 +76,48 @@ def when_I_view_the_step_by_step_page
     visit step_by_step_page_path(@step_by_step_page)
   end
 
+  def and_I_approve_the_step_by_step_with_an_optional_note
+    when_I_click_button "Approve"
+    then_the_page_should_have_title_caption "Approve step by step"
+    and_I_fill_in_an_optional_note
+    and_when_I_click_button "Yes, approve 2i"
```

Comment:
I'd go with making these be the code to actually do the thing rather than descriptive code.

For example:

```
click_on "Approve"
expect(page).to have_text("Approve step by step")
fill_in "additional_comment", with: "Please fix typo before publishing"
click_on "Yes, approve 2i"
```

which ends being more succinct and removes a layer of abstraction.